### PR TITLE
CCIP-4114 refactor token pool tests for one function per file

### PR DESCRIPTION
**Pull request by jhweintraub**
Refactor tests in the token pools folder to better align with new formatting and solhint rules. One function per file. Hybrid USDC Token pool tests were also modified in name to align with the naming convention. 

Previous Lines in the ccip.gas-snapshot: 748
Current lines: 756

### CODE REVIEW

File: contracts/src/v0.8/ccip/test/pools/USDC/USDCTokenPool.lockOrBurn.t.sol [hunk 1]
```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BUSL-1.1
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/pools/USDC/USDCTokenPool.lockOrBurn.t.sol]: The folder name should match the contract name, so not USDC but `USDCTokenPool` inside the USDC folder. It should follow the source code structure which means the hybrid USDC stuff and this should be in a USDC folder

File: contracts/src/v0.8/ccip/test/pools/BurnFromMint/BurnFromMintTokenPool.lockOrBurn.t.sol [hunk 2]
```diff
@@ -1,37 +1,23 @@
...
-  function test_Setup_Success() public view {
+  function test_setup_Success() public view {
     assertEq(address(s_burnMintERC677), address(s_pool.getToken()));
     assertEq(address(s_mockRMN), s_pool.getRmnProxy());
     assertEq(false, s_pool.getAllowListEnabled());
     assertEq(type(uint256).max, s_burnMintERC677.allowance(address(s_pool), address(s_pool)));
     assertEq("BurnFromMintTokenPool 1.5.0", s_pool.typeAndVersion());
   }
 
-  function test_PoolBurn_Success() public {
```
Comment (RensR) [inline contracts/src/v0.8/ccip/test/pools/BurnFromMint/BurnFromMintTokenPool.lockOrBurn.t.sol]: Why did you change the casing on these? The test rename comes after and these new names are not following the final naming scheme anyway?

Comment (jhweintraub) [reply contracts/src/v0.8/ccip/test/pools/BurnFromMint/BurnFromMintTokenPool.lockOrBurn.t.sol]: it was giving annoying linting errors about the case which is why but I will revert it.

Comment (RensR) [reply contracts/src/v0.8/ccip/test/pools/BurnFromMint/BurnFromMintTokenPool.lockOrBurn.t.sol]: Huh!? Solhint issues?

Comment (jhweintraub) [reply contracts/src/v0.8/ccip/test/pools/BurnFromMint/BurnFromMintTokenPool.lockOrBurn.t.sol]: no idea.

**Review by RensR — APPROVED** [review]

**Review by RyanRHall — APPROVED** [review]
